### PR TITLE
Bug/1276/edge preview broken when selecting zero

### PR DESCRIPTION
**Pull request by RomanenkoVladimir**
#Edge Preview shows all edges when slectiing 0

_**Please read the [CONTRIBUTING.md](https://github.com/MaibornWolff/codecharta/blob/main/CONTRIBUTING.md) before opening a PR.**_

closes: #1276

## Description

- Fix the bug with all edges being shown when selecting 0 for the amount.

### CODE REVIEW

File: visualization/app/codeCharta/ui/codeMap/codeMap.actions.service.ts [hunk 1]
```diff
@@ -60,7 +60,7 @@ export class CodeMapActionsService {
 		for (const edge of edges) {
 			edge.visible = EdgeVisibility.none
 
-			if (edge.attributes[edgeMetric] !== undefined) {
+			if (edge.attributes[edgeMetric] !== undefined && edgePreviewNodes !== null) {
```
Comment (BridgeAR) [inline visualization/app/codeCharta/ui/codeMap/codeMap.actions.service.ts:63]: A set may never be null :-)

```suggestion
			if (edge.attributes[edgeMetric] !== undefined) {
```

File: visualization/app/codeCharta/state/store/metricData/edgeMetricData/edgeMetricData.service.ts [hunk 2]
```diff
@@ -62,6 +62,10 @@ export class EdgeMetricDataService implements StoreSubscriber, BlacklistSubscrib
 			return keys
 		}
 
+		if (amountOfEdgePreviews === 0) {
+			return null
+		}
```
Comment (BridgeAR) [inline visualization/app/codeCharta/state/store/metricData/edgeMetricData/edgeMetricData.service.ts:67]: I guess it's nicer to return an empty array instead? In that case it's possible to combine to condition with the one above.

File: visualization/app/codeCharta/state/store/metricData/edgeMetricData/edgeMetricData.service.ts [hunk 3]
```diff
@@ -62,6 +62,10 @@ export class EdgeMetricDataService implements StoreSubscriber, BlacklistSubscrib
 			return keys
 		}
 
+		if (amountOfEdgePreviews === 0) {
```
Comment (ce-bo) [inline visualization/app/codeCharta/state/store/metricData/edgeMetricData/edgeMetricData.service.ts:65]: I would place this condition at the top of the method. If I understand it correctly, we don not need to retrieve the `nodeEdgeMetrics` in this case.
Furthermore, I would just return an empty array like in the case the `nodeEdgeMetrics` are `undefined`. Thus, no further changes would be neccessary and we can revert the change here codeMap.actions.service.ts:63.

**Review by ce-bo — COMMENTED** [review]
Could you please address my remarks? :)

File: visualization/app/codeCharta/ui/codeMap/codeMap.actions.service.ts [hunk 4]
```diff
@@ -60,7 +60,7 @@ export class CodeMapActionsService {
 		for (const edge of edges) {
 			edge.visible = EdgeVisibility.none
 
-			if (edge.attributes[edgeMetric] !== undefined && edgePreviewNodes !== null) {
```
Comment (ce-bo) [inline visualization/app/codeCharta/ui/codeMap/codeMap.actions.service.ts]: We should check for a Set initialized with an empty [], if the else-branch in line 73 implements a valid behaviour.

Comment (RomanenkoVladimir) [reply visualization/app/codeCharta/ui/codeMap/codeMap.actions.service.ts]: You mean for correctness? 
Because while we set visibility for to-edges they are empty as well, so no edges are drawn on the canvas.

Comment (RomanenkoVladimir) [reply visualization/app/codeCharta/ui/codeMap/codeMap.actions.service.ts]: I guess one could argue that the current result is incorrect, because the function shows
both if both exist
to if its the only existing direction
from else

However in an empty case the last transformation would lead to empty, which is not an expected result according to such a definition

Comment (RomanenkoVladimir) [reply visualization/app/codeCharta/ui/codeMap/codeMap.actions.service.ts]: I deleted my previous comments
The first if would catch an empty set because from == to == [] then the inner if checks it from is [] and only if it isn't it sets visibility

Comment (RomanenkoVladimir) [reply visualization/app/codeCharta/ui/codeMap/codeMap.actions.service.ts]: Missed that point as well, but now we can be sure there is no issue

Comment (ce-bo) [reply visualization/app/codeCharta/ui/codeMap/codeMap.actions.service.ts]: Thanks, there is definitely no bug :)

**Review by ce-bo — APPROVED** [review]
👍

**Review by BridgeAR — APPROVED** [review]
